src/tartiflette_asgi/_subscriptions/impl.py

```diff
@@ -37,7 +37,7 @@ async def close(self, close_code: int) -> None:
     # GraphQL engine implementation.
 
     def get_stream(self, opid: str, payload: dict) -> typing.AsyncGenerator:
-        context = {**payload.get("context", {}), **self.context}
+        context = {"req": self.websocket, **payload.get("context", {}), **self.context}
```

Comment:
Should we instead include this at the same level than we do for the HTTP request?

https://github.com/tartiflette/tartiflette-asgi/blob/9a0b41b178f5acc1934830f904ecb26a1fcae993/src/tartiflette_asgi/_endpoints.py#L64

And so change `SubscriptionEndpoint`:

https://github.com/tartiflette/tartiflette-asgi/blob/9a0b41b178f5acc1934830f904ecb26a1fcae993/src/tartiflette_asgi/_endpoints.py#L106-L108

So it does something like this…

```python
context = {"ws": websocket, **config.context}
self.protocol = GraphQLWSProtocol(..., context=context)
```

Reply:
@florimondmanca - yes including it in the `_endpoints.py` makes much more sense along with using `ws` key instead of `req`. I've pushed a change to reflect this.